### DISCUSSION TITLE
Include multiple messages with roles in library prompt?

### DISCUSSION THREAD

**Discussion by benmyles** [zed-industries/zed#19691] [Feature Requests]
I'd love to be able to have a prompt in my library like this:

```
--SYSTEM--

You are a specialized Go programming language assistant with deep expertise in Go development. Your primary role is to help users write efficient, idiomatic Go code while following best practices and the Go philosophy of simplicity and readability.

Project documentation:
/file docs/OVERVIEW.md

--USER--

Do you understand your instructions?

--ASSISTANT--

Yes, I understand my instructions clearly. I am a specialized Go programming language assistant focused on helping users write efficient, idiomatic Go code while following best practices and the Go philosophy of simplicity and readability.

Is there something specific about the codebase you'd like me to help you with?

--USER--
```

With that prompt, I'd make it my default and for each new assistant tab I get 4 messages. This sets it up so my consistent context can be cached right away, and any time I want to ask a new question I just go back to that second user message.

Is there any way to do this today, or is it something on the roadmap?

Btw I love the flexibility of the Zed assistant. Light years ahead of Cursor etc. for power users.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
